src/urllib3/contrib/securetransport.py

```diff
@@ -570,40 +570,49 @@ def recv_into(
         if nbytes is None:
             nbytes = len(buffer)
 
-        buffer = (ctypes.c_char * nbytes).from_buffer(buffer)
         processed_bytes = ctypes.c_size_t(0)
+        processed_bytes_total = 0
 
-        with self._raise_on_error():
-            result = Security.SSLRead(
-                self.context, buffer, nbytes, ctypes.byref(processed_bytes)
-            )
+        while nbytes:
```

Comment:
To avoid overrunning into always-truthy negative numbers can we use `while processed_bytes_total < nbytes:`

Reply:
I updated the condition to avoid the problem, but it is `while nbytes < 0:` now because both `nbytes` and `processed_bytes_total` change every successful iteration and `processed_bytes_total < nbytes` may become true too early.